libnetwork/drivers/bridge/internal/nftabler/network.go

```diff
@@ -19,14 +23,216 @@ func (nft *nftabler) NewNetwork(ctx context.Context, nc firewaller.NetworkConfig
 		fw:     nft,
 		config: nc,
 	}
+	defer func() {
+		if retErr != nil {
+			if err := n.DelNetworkLevelRules(ctx); err != nil {
+				log.G(ctx).WithError(err).Error("Ignoring cleanup error")
+			}
+		}
+	}()
+
+	if n.fw.config.IPv4 {
+		if err := n.configure(ctx, nft.table4, n.config.Config4); err != nil {
+			return nil, err
+		}
+	}
+	if n.fw.config.IPv6 {
+		if err := n.configure(ctx, nft.table6, n.config.Config6); err != nil {
+			return nil, err
+		}
+	}
 	return n, nil
 }
 
+func (n *network) configure(ctx context.Context, table nftables.TableRef, conf firewaller.NetworkConfigFam) error {
+	ctx, span := otel.Tracer("").Start(ctx, spanPrefix+".newNetwork."+string(table.Family()))
+	defer span.End()
+
+	if !conf.Prefix.IsValid() {
+		return nil
+	}
+
+	// Filter chain
+	fwdInChain := table.Chain(chainFilterFwdIn(n.config.IfName))
+	fwdOutChain := table.Chain(chainFilterFwdOut(n.config.IfName))
+	if err := table.InterfaceVMap(filtFwdInVMap).AddElement(n.config.IfName, "jump "+chainFilterFwdIn(n.config.IfName)); err != nil {
+		return fmt.Errorf("adding filter-forward jump for %s to %q: %w", conf.Prefix, chainFilterFwdIn(n.config.IfName), err)
+	}
+	if err := table.InterfaceVMap(filtFwdOutVMap).AddElement(n.config.IfName, "jump "+chainFilterFwdOut(n.config.IfName)); err != nil {
+		return fmt.Errorf("adding filter-forward jump for %s to %q: %w", conf.Prefix, chainFilterFwdOut(n.config.IfName), err)
+	}
+
+	// NAT chain
+	natPostroutingIn := table.Chain(chainNatPostRtIn(n.config.IfName))
+	if err := table.InterfaceVMap(natPostroutingInVMap).AddElement(n.config.IfName, "jump "+chainNatPostRtIn(n.config.IfName)); err != nil {
+		return fmt.Errorf("adding postrouting ingress jump for %s to %q: %w", conf.Prefix, chainNatPostRtIn(n.config.IfName), err)
+	}
+	natPostroutingOut := table.Chain(chainNatPostRtOut(n.config.IfName))
+	if err := table.InterfaceVMap(natPostroutingOutVMap).AddElement(n.config.IfName, "jump "+chainNatPostRtOut(n.config.IfName)); err != nil {
+		return fmt.Errorf("adding postrouting egress jump for %s to %q: %w", conf.Prefix, chainNatPostRtOut(n.config.IfName), err)
+	}
+
+	// Conntrack
+	if err := fwdInChain.AppendRule(initialRuleGroup, "ct state established,related counter accept"); err != nil {
```

Comment:
I think we should add a rule that drops packets marked as `invalid`. We don't have one in the current iptables implementation, but that's probably a mistake.